server/src/main/java/org/opensearch/index/translog/TranslogManager.java

```diff
@@ -33,6 +33,12 @@ public interface TranslogManager {
      */
     int recoverFromTranslog(TranslogRecoveryRunner translogRecoveryRunner, long localCheckpoint, long recoverUpToSeqNo) throws IOException;
 
+    /**
+     * Creates a new history snapshot from the translog file instead of the lucene index.
```

Comment:
I think we may want to deprecate this as we're moving toward segment replication and segment based recovery. /cc @mch2

Reply:
hmm I think this depends on the configuration options we want to provide for users with CCR + segrep.  If a user does not want to depend on remote store there will likely be cases where ops based replication wins out over node based primary->primary segment replication for larger clusters.  I think for now it may be advantageous to provide this as an additional configuration option, even if not the default in the near future.

Reply:
Can we add an assertion that this method will be invoked only for logical replication, which we could continue to support, so that when we switch to making segrep the default we can safely deprecate this method

Reply:
Added an assertion so that this method would be invoked only for logical rep & local store indices.